lib/ansible/modules/cloud/amazon/redshift.py

```diff
@@ -327,12 +340,23 @@ def delete_cluster(module, redshift):
     redshift: authenticated redshift connection object
     """
 
-    identifier   = module.params.get('identifier')
-    wait         = module.params.get('wait')
-    wait_timeout = module.params.get('wait_timeout')
+    identifier                        = module.params.get('identifier')
+    wait                              = module.params.get('wait')
+    wait_timeout                      = module.params.get('wait_timeout')
+    skip_final_cluster_snapshot       = module.params.get('skip_final_cluster_snapshot')
+    final_cluster_snapshot_identifier = module.params.get('final_cluster_snapshot_identifier')
+
+    if skip_final_cluster_snapshot and final_cluster_snapshot_identifier is not None:
+        module.fail_json(msg="Final snapshot is skipped but a snapshot id was specified")
+    if not skip_final_cluster_snapshot and final_cluster_snapshot_identifier is None:
+        module.fail_json(msg="Snapshot identifier (final_cluster_snapshot_identifier) must be provided if skip_final_cluster_snapshot is false.")
```

Comment:
Instead of enforcing in code, you can use "required_if" on the argspec to enforce this.